packages/frontend-v2/hooks/useFetchSearchCourse.ts

```diff
@@ -0,0 +1,61 @@
+import useSWR, { SWRResponse } from "swr";
+import { SearchAPI } from "@graduate/api-client";
+import { AxiosError } from "axios";
+import { ScheduleCourse2 } from "@graduate/common";
+
+type FetchCourseResponse = Omit<
+  SWRResponse<ScheduleCourse2<null>, AxiosError | Error>,
+  "data" | "mutate"
+>;
+
+type FetchCourseReturn = FetchCourseResponse & {
+  course?: ScheduleCourse2<null> | null;
+  isLoading: boolean;
+};
+
+/**
+ * @param subject - The type of class that we are fetching from SearchNEU
+ * @param classId - The identification number of the class as a string
+ */
+export function useFetchSearchCourse(
```

Comment:
let's remove "search" from all our hooks, it's kinda confusing(like are we searching with this hook?) and besides, fetching from SearchNEU using the SearchClient is an implementation detail of the hook that isn't very useful. 

When using the hooks, all we care about is the returned data(courses). How we get them could evolve. Never know, down the road we may not even fetch from SearchNEU 🤯